File: SeoulTools/Data/Config/UnitTests/SlimCS/Scripts/Engine/UI/Animation2DNetwork.lua
-- Placeholder API for the native ScriptUIAnimation2DNetworkInstance.
--
-- Copyright (c) Demiurge Studios, Inc.
-- 
-- This source code is licensed under the MIT license.
-- Full license details can be found in the LICENSE file
-- in the root directory of this source tree.







local Animation2DNetwork = class('Animation2DNetwork', 'DisplayObject')


-- Utility to check for ready state - all methods in this instance
-- depend on IsReady() and we raise an error if they're called
-- otherwise (the Animation2DNetworkAsync utility is used
-- to enforce this in practice).
local function CheckReady(self)


	if not self.m_udNativeInstance:IsReady() then

		error("Animation2DNetwork instance '" .. tostring(self:GetName()) .. "' is not ready.", 1)
	end
end




function Animation2DNetwork:get_ActivePalette()

	--[[CheckReady(self)]]
	return self.m_udNativeInstance:GetActivePalette()
end

function Animation2DNetwork:set_ActivePalette(value)

	--[[CheckReady(self)]]
	self.m_udNativeInstance:SetActivePalette(value)
end




function Animation2DNetwork:get_ActiveSkin()

	--[[CheckReady(self)]]
	return self.m_udNativeInstance:GetActiveSkin()
end

function Animation2DNetwork:set_ActiveSkin(value)

	--[[CheckReady(self)]]
	self.m_udNativeInstance:SetActiveSkin(value)
end


function Animation2DNetwork:GetActiveStatePath()

	--[[CheckReady(self)]]
	return self.m_udNativeInstance:GetActiveStatePath()
end



function Animation2DNetwork:get_IsReady()

	return self.m_udNativeInstance:IsReady()
end


function Animation2DNetwork:AddBoneAttachment(iIndex, attachment)

	--[[CheckReady(self)]]
	self.m_udNativeInstance:AddBoneAttachment(
		iIndex,
		attachment:get_NativeInstance())
end

function Animation2DNetwork:AddTimestepOffset(fTimestepOffset)

	--[[CheckReady(self)]]
	self.m_udNativeInstance:AddTimestepOffset(fTimestepOffset)
end

function Animation2DNetwork:AllDonePlaying()

	--[[CheckReady(self)]]
	return self.m_udNativeInstance:AllDonePlaying()
end



function Animation2DNetwork:get_CurrentMaxTime()

	--[[CheckReady(self)]]
	return self.m_udNativeInstance:GetCurrentMaxTime()
end


function Animation2DNetwork:GetBoneIndex(sName)

	--[[CheckReady(self)]]
	return self.m_udNativeInstance:GetBoneIndex(sName)
end

function Animation2DNetwork:GetBonePositionByName(sName)

	--[[CheckReady(self)]]
	return self.m_udNativeInstance:GetBonePositionByName(sName)
end

function Animation2DNetwork:GetBonePositionByIndex(iIndex)

	--[[CheckReady(self)]]
	return self:GetBonePositionByIndex(iIndex)
end

function Animation2DNetwork:GetLocalBonePositionByName(sName)

	--[[CheckReady(self)]]
	return self.m_udNativeInstance:GetLocalBonePositionByName(sName)
end

function Animation2DNetwork:GetLocalBonePositionByIndex(iIndex)

	--[[CheckReady(self)]]
	return self.m_udNativeInstance:GetLocalBonePositionByIndex(iIndex)
end

function Animation2DNetwork:GetLocalBoneScaleByIndex(iIndex)

	--[[CheckReady(self)]]
	return self.m_udNativeInstance:GetLocalBoneScaleByIndex(iIndex)
end

function Animation2DNetwork:GetLocalBoneScaleByName(sName)

	--[[CheckReady(self)]]
	return self.m_udNativeInstance:GetLocalBoneScaleByName(sName)
end

function Animation2DNetwork:GetWorldSpaceBonePositionByIndex(iIndex)

	--[[CheckReady(self)]]
	return self.m_udNativeInstance:GetWorldSpaceBonePositionByIndex(iIndex)
end

function Animation2DNetwork:GetWorldSpaceBonePositionByName(sName)

	--[[CheckReady(self)]]
	return self.m_udNativeInstance:GetWorldSpaceBonePositionByName(sName)
end

function Animation2DNetwork:GetTimeToEvent(sEventName)

	--[[CheckReady(self)]]
	return self.m_udNativeInstance:GetTimeToEvent(sEventName)
end



function Animation2DNetwork:get_IsInStateTransition()

	--[[CheckReady(self)]]
	return self.m_udNativeInstance:IsInStateTransition()
end


function Animation2DNetwork:SetCastShadow(bCast)

	--[[CheckReady(self)]]
	self.m_udNativeInstance:SetCastShadow(bCast)
end

function Animation2DNetwork:SetCondition(sName, bValue)

	--[[CheckReady(self)]]
	self.m_udNativeInstance:SetCondition(sName, bValue)
end

function Animation2DNetwork:SetShadowOffset(fX, fY)

	--[[CheckReady(self)]]
	self.m_udNativeInstance:SetShadowOffset(fX, fY)
end

function Animation2DNetwork:SetParameter(sName, fValue)

	--[[CheckReady(self)]]
	self.m_udNativeInstance:SetParameter(sName, fValue)
end

function Animation2DNetwork:TriggerTransition(sName)

	--[[CheckReady(self)]]
	self.m_udNativeInstance:TriggerTransition(sName)
end



function Animation2DNetwork:set_VariableTimeStep(value)

	--[[CheckReady(self)]]
	self.m_udNativeInstance:SetVariableTimeStep(value)
end

return Animation2DNetwork

-- /#if SEOUL_WITH_ANIMATION_2D

File: SeoulTools/Data/Config/UnitTests/SlimCS/ScriptsCS/Engine/UI/Animation2DNetwork.cs
// Placeholder API for the native ScriptUIAnimation2DNetworkInstance.
//
// Copyright (c) Demiurge Studios, Inc.
// 
// This source code is licensed under the MIT license.
// Full license details can be found in the LICENSE file
// in the root directory of this source tree.

using static SlimCS;

using System.Diagnostics;

#if SEOUL_WITH_ANIMATION_2D

public sealed class Animation2DNetwork : DisplayObject
{
	#region Private member
	// Utility to check for ready state - all methods in this instance
	// depend on IsReady() and we raise an error if they're called
	// otherwise (the Animation2DNetworkAsync utility is used
	// to enforce this in practice).
	[Conditional("DEBUG")]
	void CheckReady()
	{
		if (!dyncast<Native.ScriptUIAnimation2DNetworkInstance>(m_udNativeInstance).IsReady())
		{
			error($"Animation2DNetwork instance '{GetName()}' is not ready.", 1);
		}
	}
	#endregion

	public string ActivePalette
	{
		get
		{
			CheckReady();
			return dyncast<Native.ScriptUIAnimation2DNetworkInstance>(m_udNativeInstance).GetActivePalette();
		}

		set
		{
			CheckReady();
			dyncast<Native.ScriptUIAnimation2DNetworkInstance>(m_udNativeInstance).SetActivePalette(value);
		}
	}

	public string ActiveSkin
	{
		get
		{
			CheckReady();
			return dyncast<Native.ScriptUIAnimation2DNetworkInstance>(m_udNativeInstance).GetActiveSkin();
		}

		set
		{
			CheckReady();
			dyncast<Native.ScriptUIAnimation2DNetworkInstance>(m_udNativeInstance).SetActiveSkin(value);
		}
	}

	public (string, double) GetActiveStatePath()
	{
		CheckReady();
		return dyncast<Native.ScriptUIAnimation2DNetworkInstance>(m_udNativeInstance).GetActiveStatePath();
	}

	public bool IsReady
	{
		get
		{
			return dyncast<Native.ScriptUIAnimation2DNetworkInstance>(m_udNativeInstance).IsReady();
		}
	}

	public void AddBoneAttachment(double iIndex, DisplayObject attachment)
	{
		CheckReady();
		dyncast<Native.ScriptUIAnimation2DNetworkInstance>(m_udNativeInstance).AddBoneAttachment(
			iIndex,
			attachment.NativeInstance);
	}

	public void AddTimestepOffset(double fTimestepOffset)
	{
		CheckReady();
		dyncast<Native.ScriptUIAnimation2DNetworkInstance>(m_udNativeInstance).AddTimestepOffset(fTimestepOffset);
	}

	public (bool, bool) AllDonePlaying()
	{
		CheckReady();
		return dyncast<Native.ScriptUIAnimation2DNetworkInstance>(m_udNativeInstance).AllDonePlaying();
	}

	public double CurrentMaxTime
	{
		get
		{
			CheckReady();
			return dyncast<Native.ScriptUIAnimation2DNetworkInstance>(m_udNativeInstance).GetCurrentMaxTime();
		}
	}

	public double GetBoneIndex(string sName)
	{
		CheckReady();
		return dyncast<Native.ScriptUIAnimation2DNetworkInstance>(m_udNativeInstance).GetBoneIndex(sName);
	}

	public (double, double) GetBonePositionByName(string sName)
	{
		CheckReady();
		return dyncast<Native.ScriptUIAnimation2DNetworkInstance>(m_udNativeInstance).GetBonePositionByName(sName);
	}

	public (double, double) GetBonePositionByIndex(double iIndex)
	{
		CheckReady();
		return GetBonePositionByIndex(iIndex);
	}

	public (double, double) GetLocalBonePositionByName(string sName)
	{
		CheckReady();
		return dyncast<Native.ScriptUIAnimation2DNetworkInstance>(m_udNativeInstance).GetLocalBonePositionByName(sName);
	}

	public (double, double) GetLocalBonePositionByIndex(double iIndex)
	{
		CheckReady();
		return dyncast<Native.ScriptUIAnimation2DNetworkInstance>(m_udNativeInstance).GetLocalBonePositionByIndex(iIndex);
	}

	public (double, double) GetLocalBoneScaleByIndex(double iIndex)
	{
		CheckReady();
		return dyncast<Native.ScriptUIAnimation2DNetworkInstance>(m_udNativeInstance).GetLocalBoneScaleByIndex(iIndex);
	}

	public (double, double) GetLocalBoneScaleByName(string sName)
	{
		CheckReady();
		return dyncast<Native.ScriptUIAnimation2DNetworkInstance>(m_udNativeInstance).GetLocalBoneScaleByName(sName);
	}

	public (double, double) GetWorldSpaceBonePositionByIndex(double iIndex)
	{
		CheckReady();
		return dyncast<Native.ScriptUIAnimation2DNetworkInstance>(m_udNativeInstance).GetWorldSpaceBonePositionByIndex(iIndex);
	}

	public (double, double) GetWorldSpaceBonePositionByName(string sName)
	{
		CheckReady();
		return dyncast<Native.ScriptUIAnimation2DNetworkInstance>(m_udNativeInstance).GetWorldSpaceBonePositionByName(sName);
	}

	public double? GetTimeToEvent(string sEventName)
	{
		CheckReady();
		return dyncast<Native.ScriptUIAnimation2DNetworkInstance>(m_udNativeInstance).GetTimeToEvent(sEventName);
	}

	public bool IsInStateTransition
	{
		get
		{
			CheckReady();
			return dyncast<Native.ScriptUIAnimation2DNetworkInstance>(m_udNativeInstance).IsInStateTransition();
		}
	}

	public void SetCastShadow(bool bCast)
	{
		CheckReady();
		dyncast<Native.ScriptUIAnimation2DNetworkInstance>(m_udNativeInstance).SetCastShadow(bCast);
	}

	public void SetCondition(string sName, bool bValue)
	{
		CheckReady();
		dyncast<Native.ScriptUIAnimation2DNetworkInstance>(m_udNativeInstance).SetCondition(sName, bValue);
	}

	public void SetShadowOffset(double fX, double fY)
	{
		CheckReady();
		dyncast<Native.ScriptUIAnimation2DNetworkInstance>(m_udNativeInstance).SetShadowOffset(fX, fY);
	}

	public void SetParameter(string sName, double fValue)
	{
		CheckReady();
		dyncast<Native.ScriptUIAnimation2DNetworkInstance>(m_udNativeInstance).SetParameter(sName, fValue);
	}

	public void TriggerTransition(string sName)
	{
		CheckReady();
		dyncast<Native.ScriptUIAnimation2DNetworkInstance>(m_udNativeInstance).TriggerTransition(sName);
	}

	public bool VariableTimeStep
	{
		set
		{
			CheckReady();
			dyncast<Native.ScriptUIAnimation2DNetworkInstance>(m_udNativeInstance).SetVariableTimeStep(value);
		}
	}
}

#endif // /#if SEOUL_WITH_ANIMATION_2D
